apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx

```diff
@@ -239,6 +243,10 @@ const ContentWrapper = ({ isLoading, isBlocking = true, children }: ContentWrapp
     return router.pathname.endsWith('[ref]') ? <LoadingState /> : <Loading />
   }
 
+  if (isRestarting) {
+    return <RestartingState />
```

Comment:
just wanna double check - any updates needed in RestartServerButton line 78 onRestartSuccess? Now that we've got a specific status for restarting

Reply:
yep good call – I'll update to that restarting status, although the backend will need to return the restarting status from the `ref/status` endpoint. It'll still work for now just not be as smooth!